File: 700_search_in_a_binary_search_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        def f(node):
            if not node:
                return
            elif node.val == val:
                return node
            elif val < node.val:
                return f(node.left)
            return f(node.right)

        return f(root)

File: JavaScript/700_search_in_a_binary_search_tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
    function f(node) {
        if (!node) {
            return null;
        } else if (node.val === val) {
            return node;
        } else if (node.val < val) {
            return f(node.right);
        } else {
            return f(node.left);
        }
    }
    return f(root);
};

